From b0703c8eadb371cb28fbcc27a89ab59f8ab68ad8 Mon Sep 17 00:00:00 2001
From: traxar <trxr2a@gmail.com>
Date: Sat, 15 Nov 2025 16:39:39 +0900
Subject: [PATCH 063/133] src/codegen/spirv/CodeGen.zig: add missing unary Ops
X-Developer-Signature: v=1; a=openpgp-sha256; l=868; i=xtex@astrafall.org;
 h=from:subject; bh=CN12+yA7esdce5UquSgmYybT4NrpfrqWcfLBt2QHvPY=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78pitSsSaQ5wnfTzwmetMvItEjcfmHwWd7+xYCla5X
 2C0PHWnYyILgxgXg6WYIkuRYYM3q046v+iyclmYOaxMIEOkRRoYGBgYWBj4chPzSo10jPRMtQ31
 DA11jHSMGLg4BWCqo/Yz/NPMm/4+VlDuR+esuZJOM5gSVd1kCkOVBGND2O6bvviavomRYcYdydK
 TRlWyp+bt28Z8MezmtltbV+8s03ff/Ms/o+fOK2YA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

(cherry picked from commit 868413007f8cec8bed6d4bef446c0fa9cdaf6da0)
---
 src/codegen/spirv/CodeGen.zig | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/codegen/spirv/CodeGen.zig b/src/codegen/spirv/CodeGen.zig
index 0edcfa591905..e85dde2ea476 100644
--- a/src/codegen/spirv/CodeGen.zig
+++ b/src/codegen/spirv/CodeGen.zig
@@ -2177,6 +2177,14 @@ const UnaryOp = enum {
                 .ceil => .Ceil,
                 .trunc => .Trunc,
                 .round => .Round,
+                .sin => .Sin,
+                .cos => .Cos,
+                .tan => .Tan,
+                .sqrt => .Sqrt,
+                .exp => .Exp,
+                .exp2 => .Exp2,
+                .log => .Log,
+                .log2 => .Log2,
                 else => return null,
             })),
             else => unreachable,
-- 
2.52.0

